crates/sui-protocol-config/src/lib.rs

```diff
@@ -179,71 +179,100 @@ impl Default for Chain {
     }
 }
 
+fn bool_default() -> bool {
+    false
+}
+
+fn consensus_transaction_ordering_default() -> ConsensusTransactionOrdering {
+    ConsensusTransactionOrdering::None
+}
+
+fn zklogin_supported_providers_default() -> BTreeSet<String> {
+    BTreeSet::new()
+}
+
 pub struct Error(pub String);
 
 /// Records on/off feature flags that may vary at each protocol version.
-#[derive(Default, Clone, Serialize, Debug, ProtocolConfigFeatureFlagsGetters)]
+#[derive(Default, Clone, Serialize, Deserialize, Debug, ProtocolConfigFeatureFlagsGetters)]
 struct FeatureFlags {
```

Comment:
These changes are needed so we can deserialize ProtocolConfig. Alternatives are welcomed

Reply:
Why do we need to deserialize ProtocolConfig?

Reply:
The snapshot ships with the exact protocol config. We could get away with using the protocol version and fetching as needed but I wanted the file to be as self contained as possible.

Reply:
That's interesting. I actually think we shouldn't snapshot the protocol config. It's not stored on-chain but part of the software code, obtained from protocol version. This will allow us to catch a mistake where we messed up our protocol config.